airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -59,7 +59,55 @@
 FARGATE_FULL_NAME = "AWS Fargate profiles"
 
 
-class EksCreateClusterOperator(BaseOperator):
+class BaseEksCreateOperator(BaseOperator):
+    """
+    Base class for EKS Operators that create cluster resources. This includes
+    EksCreateClusterOperator, EksCreateNodeGroupOperator and EksCreateFargateProfileOperator.
+
+    """
+
+    def __init__(self, **kwargs):
+        self.eks_hook = EksHook(aws_conn_id=self.aws_conn_id, region_name=self.region)
+        super().__init__(**kwargs)
```

Comment:
Any reason the hook is instantiated before the `super()` call? I believe the majority preference is to do it after the super call. (I don’t think the style matters here, but it may introduce unnecessary confusion.)

Reply:
No particular reason. I didn't really pay attention to the order :sweat_smile: